### PR TITLE
`oscillator-overlap`: Make consistent with `oscillator`

### PR DESCRIPTION
**Pull request by BenjaminRodenberg**
Updates `oscillator-overlap` to use time stepping schemes and typing like in `oscillator`

Checklist:

- [x] I added a summary of any user-facing changes (compared to the last release) in the `changelog-entries/<PRnumber>.md`.
- [x] I will remember to squash-and-merge, providing a useful summary of the changes of this PR.

### CODE REVIEW

File: oscillator-overlap/solver-python/oscillator.py [hunk 1]
```diff
@@ -68,7 +74,7 @@ class Participant(Enum):
 
 vertex = np.zeros(dimensions)
 read_data = np.zeros(num_vertices)
-write_data = connecting_spring.k * u0 * np.ones(num_vertices)
```
Comment (BenjaminRodenberg) [inline oscillator-overlap/solver-python/oscillator.py:71]: This is actually a bug: In `oscillator` we exchange forces. Writing displacement * spring stiffness is correct. Compare https://github.com/precice/tutorials/blob/dea77cbfdbc3d5bdc0df1e745bbc3c3c5ed2b2d4/oscillator/solver-python/oscillator.py#L77. In this case, i.e. `oscillator-overlap` we have to write displacements. The multiplication with `connecting_spring.k` is incorrect.

File: changelog-entries/597.md [hunk 2]
```diff
@@ -0,0 +1 @@
+- Offer Runge-Kutta 4 scheme and black-box time-stepping with RadauIIA scheme in oscillator-overlap case. Demonstrates use of time interpolation and dense output.
```
Comment (uekerman) [inline changelog-entries/597.md]:
```suggestion
- Added Runge-Kutta 4 scheme and black-box time-stepping with RadauIIA scheme as options to oscillator-overlap case. This demonstrates use of time interpolation and dense output.
```

**Review by uekerman — APPROVED** [review]
Makes sense to me. I did not look into the details and did not run the case.

File: oscillator-overlap/solver-python/oscillator.py [hunk 3]
```diff
@@ -6,13 +6,10 @@
...
 import os
+from typing import Type
 
 import problemDefinition
-
-
-class Scheme(Enum):
-    NEWMARK_BETA = "Newmark_beta"
-    GENERALIZED_ALPHA = "generalized_alpha"
+from timeSteppers import TimeStepper, TimeSteppingSchemes, GeneralizedAlpha, RungeKutta4, RadauIIA
```
Comment (NiklasVin) [inline oscillator-overlap/solver-python/oscillator.py:12]: The `timeSteppers` module is missing.

File: oscillator-overlap/solver-python/oscillator.py [hunk 4]
```diff
@@ -22,20 +19,30 @@ class Participant(Enum):
 
 parser = argparse.ArgumentParser()
 parser.add_argument("participantName", help="Name of the solver.", type=str, choices=[p.value for p in Participant])
-parser.add_argument("-ts", "--time-stepping", help="Time stepping scheme being used.", type=str,
-                    choices=[s.value for s in Scheme], default=Scheme.NEWMARK_BETA.value)
+parser.add_argument(
```
Comment (NiklasVin) [inline oscillator-overlap/solver-python/oscillator.py:22]: This comment is more of a general usability concern of this feature : Assuming the user wants to use a different time stepping scheme but wants to use the `run.sh` files only, then, as far as I can see, the time stepping scheme cannot be changed.

**Review by NiklasVin — CHANGES_REQUESTED** [review]

File: oscillator-overlap/solver-python/oscillator.py (hunk 3 above)
Comment (BenjaminRodenberg) [reply oscillator-overlap/solver-python/oscillator.py:12]: Fixed. Thanks for checking!

File: oscillator-overlap/solver-python/oscillator.py (hunk 4 above)
Comment (BenjaminRodenberg) [reply oscillator-overlap/solver-python/oscillator.py:22]: True. However, in `tutorials/oscillator` we also do not offer the option to select a different time stepping scheme using the `run.sh` script. Therefore, I would keep it as it is and not add a new feature here.

**Review by NiklasVin — APPROVED** [review]
Now everything works 👍

**Pull request merged**